substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/debugentry/ClassEntry.java

```diff
@@ -280,9 +250,110 @@ public boolean includesDeoptTarget() {
 
     public String getCachePath() {
         if (fileEntry != null) {
-            return fileEntry.getCachePath();
-        } else {
+            Path cachePath = fileEntry.getCachePath();
+            if (cachePath != null) {
+                return cachePath.toString();
+            }
+        }
+        return "";
+    }
+
+    private void processInterface(String interfaceName, DebugInfoBase debugInfoBase, DebugContext debugContext) {
+        debugContext.log("typename %s adding interface %s\n", typeName, interfaceName);
+        ClassEntry entry = debugInfoBase.lookupClassEntry(TypeEntry.canonicalize(interfaceName));
+        assert entry instanceof InterfaceClassEntry;
+        InterfaceClassEntry interfaceClassEntry = (InterfaceClassEntry) entry;
+        interfaces.add(interfaceClassEntry);
+        interfaceClassEntry.addImplementor(this, debugContext);
+    }
+
+    protected void processMethod(DebugMethodInfo debugMethodInfo, DebugInfoBase debugInfoBase, DebugContext debugContext) {
+        String methodName = debugInfoBase.uniqueDebugString(debugMethodInfo.name());
+        String resultTypeName = TypeEntry.canonicalize(debugMethodInfo.valueType());
+        int modifiers = debugMethodInfo.modifiers();
+        List<String> paramTypes = debugMethodInfo.paramTypes();
+        List<String> paramNames = debugMethodInfo.paramNames();
+        assert paramTypes.size() == paramNames.size();
+        int paramCount = paramTypes.size();
+        debugContext.log("typename %s adding %s method %s %s(%s)\n",
+                        typeName, memberModifiers(modifiers), resultTypeName, methodName, formatParams(paramTypes, paramNames));
+        TypeEntry resultType = debugInfoBase.lookupTypeEntry(resultTypeName);
+        TypeEntry[] paramTypeArray = new TypeEntry[paramCount];
+        String[] paramNameArray = new String[paramCount];
+        int idx = 0;
+        for (String paramTypeName : paramTypes) {
+            TypeEntry paramType = debugInfoBase.lookupTypeEntry(TypeEntry.canonicalize(paramTypeName));
+            paramTypeArray[idx++] = paramType;
+        }
+        paramNameArray = paramNames.toArray(paramNameArray);
+        String fileName = debugMethodInfo.fileName();
+        Path filePath = debugMethodInfo.filePath();
+        Path cachePath = debugMethodInfo.cachePath();
+        // n.b. the method file may differ from the owning class file when the method is a
```

Comment:
`/* ... */`

Reply:
Done. The only remaining // comments are to exclude generation of speculative/currently unavailable field declarations for abbrev record in DwarfAbbrevSectionImpl.java. I really want to leave those as they are place-holders for possible future upgrades.